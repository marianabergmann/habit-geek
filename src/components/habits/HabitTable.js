import React from "react";

import HabitHeader from "./HabitHeader";
import HabitRows from "./HabitRows";
import "./HabitTable.scss";

const HabitTable = ({ habits, toggleDayHabit, isReadOnly }) => {
  return (
    <div className="mb-3 dashboard__table">
      <HabitHeader></HabitHeader>
      {
        <HabitRows
          habits={habits}
          toggleDayHabit={toggleDayHabit}
          isReadOnly={isReadOnly}
        ></HabitRows>
      }
    </div>
  );
};

export default HabitTable;
