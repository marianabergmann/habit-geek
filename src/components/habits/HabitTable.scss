.habit {
  &__table {
    display: flex;
    flex-direction: column;
    justify-content: space-around;
  }

  &__row {
    display: flex;
  }

  &__cell-habit-container {
    width: 140px;
  }

  &__cell-frequency-container {
    flex-grow: 1;
    display: flex;
    justify-content: space-around;
  }

  &__cell {
    width: 60px;
    text-align: center;

    .toggle {
      margin: 5px 0 5px 7px;
    }
  }
}

@media (max-width: 576px) {
  .habit {
    &__table {
      max-width: 400px;
      margin: auto;
    }

    &__row {
      flex-direction: column;
      margin-bottom: 5px;
    }

    &__cell-habit-container {
      width: 100%;
      text-align: center;
    }

    &__cell {
      font-size: 1rem;
    }
  }
}
