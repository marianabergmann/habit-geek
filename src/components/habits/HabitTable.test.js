import React from "react";
import { shallow } from "enzyme";
import HabitTable from "./HabitTable";

describe("HabitTable component", () => {
  it("Renders static elements", () => {
    const habitTable = shallow(<HabitTable />);
    expect(habitTable).toMatchSnapshot();
  });
});
