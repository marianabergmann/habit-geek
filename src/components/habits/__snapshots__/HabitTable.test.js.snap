// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HabitTable component Renders static elements 1`] = `
<div
  className="mb-3 dashboard__table"
>
  <HabitHeader />
  <HabitRows />
</div>
`;
